File: hub-ue/apps/dashboard/src/topics.ts
export const TOPICS = [
  "experience.lifecycle",
  "experience.marker",
  "unreal.state",
  "unreal.commands",
  "hrv.raw",
  "hrv.processed",
  "ecg.raw",
  "imu.accelerometer.raw",
  "temperature.raw",
  "eeg.raw",
  "eeg.processed",
  "biofeedback.events",
  "ai.input",
  "ai.output",
  "logger.events",
  "system.events",
] as const;

export type Topic = (typeof TOPICS)[number];

File: hub-ue/apps/hub/src/biofeedback_hub/schemas/topics.py
from __future__ import annotations

from enum import Enum


class Topic(str, Enum):
    EXPERIENCE_LIFECYCLE = "experience.lifecycle"
    EXPERIENCE_MARKER = "experience.marker"
    UNREAL_STATE = "unreal.state"
    UNREAL_COMMANDS = "unreal.commands"
    HRV_RAW = "hrv.raw"
    HRV_PROCESSED = "hrv.processed"
    EEG_RAW = "eeg.raw"
    EEG_PROCESSED = "eeg.processed"
    BIOFEEDBACK_EVENTS = "biofeedback.events"
    AI_INPUT = "ai.input"
    AI_OUTPUT = "ai.output"
    LOGGER_EVENTS = "logger.events"
    SYSTEM_EVENTS = "system.events"
